lib/poker/hand.rb

```diff
@@ -0,0 +1,88 @@
+module Poker
+  class Hand
+
+    def initialize(cards)
+      @cards = cards
+      @sets = sets
```

Comment:
The method in which this is memoized is a bit weird. We should contain this logic in the method below.